File: apps/portal-client/src/types/leaderboard.ts
// src/types/leaderboard.ts

/**
 * Um único item do ranking.
 */
export interface LeaderboardEntry {
  user_id: string;   // UUID do usuário
  name: string;      // nome do usuário
  points: number;    // pontos acumulados
}

/**
 * Paginação genérica do ranking.
 */
export interface PaginatedLeaderboard {
  total: number;                 // total de registros no ranking
  skip: number;                  // quantos registros foram pulados
  limit: number;                 // quantos registros vieram
  items: LeaderboardEntry[];     // lista de entradas do ranking
  generated_at: string;          // timestamp ISO de quando foi gerado
}

File: apps/backend/app/schemas/leaderboard.py
# app/schemas/leaderboard.py
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel, ConfigDict

class LeaderboardEntry(BaseModel):
    user_id: UUID
    name: str
    points: int

class PaginatedLeaderboard(BaseModel):
    total: int
    skip: int
    limit: int
    items: list[LeaderboardEntry]
    generated_at: datetime

    model_config = ConfigDict(from_attributes=True)
